### ISSUE TITLE
"Missing required positional argument" for default field

### ISSUE THREAD

**Issue by fortierq** [smarie/python-pyfields#81]
Hello,

In the following code, I get: TypeError: __init__() missing 1 required positional argument: 'a'

```
from pyfields import autoclass, field

@autoclass
class A:
    a: int = field(default = 1)

@autoclass
class B(A):
    b = field()
    
B(b=3)
```

It works when I add a type hint to b (b : int = field()). 
Is it a bug or a feature?
pyfields version: 1.6.0

**Comment by smarie** [comment]
Thanks @fortierq for opening this ! Indeed I am able to reproduce the issue.
It seems that the issue comes from the fact that I misunderstood how `B.__annotations__` works: apparently it shows an aggregation of all annotations including inherited ones. So it contains `{'a': int}`, For this reason `autofields` does not work correctly, as it does not assume inherited annotations.

I need to investigate how to fix this.

**Comment by fortierq** [comment]
Thanks for your answer.
Not sure I understand the issue, but I am definitely interested by looking at the code when I find the time.
I like the default_factory option, which seems missing from dataclasses.

**Comment by smarie** [comment]
I found a good way to do the check [here](https://github.com/python-attrs/attrs/issues/288#issuecomment-347014841).
That should now be easy to fix, I'll do it in the upcoming days. Thanks again !

**Comment by smarie** [comment]
This should be fixed in 1.6.1, that will be available in a few minutes. Thanks again @fortierq  !

**Comment by fortierq** [comment]
Great! Thanks for your work.
